services/fedora/fedora.service.js

```diff
@@ -0,0 +1,49 @@
+'use strict'
+
+const Joi = require('joi')
+const { renderVersionBadge } = require('../version')
+const { BaseJsonService } = require('..')
+
+const schema = Joi.object({
+  version: Joi.string().required(),
+}).required()
+
+module.exports = class Fedora extends BaseJsonService {
+  static get category() {
+    return 'version'
+  }
+
+  static get route() {
+    return {
+      base: 'fedora/v',
+      pattern: ':branch/:packageName',
```

Comment:
Is it possible for `branch` to contain a `/` character (like a `feature/abc` for branch)?

Reply:
As far as I know, no. The branch name occurs in various filenames in their distros, and that wouldn't work with a slash in it. I find it highly unlikely that they'd ever go through all the work that would make it possible. Their own API uses the branch as a path component as well, see https://apps.fedoraproject.org/mdapi/

Reply:
In almost all of our badges, we put the branch after the package name. If it's optional, I'm inclined to do the same thing here. We usually try to stick with consistency of our own conventions over matching the conventions of other projects and services. Every service does things a little differently, and it's helpful to give users a consistent experience.

Reply:
At least [Debian] and [Ubuntu] already have the optional branch before the package name, and [ArchLinux] has that kind of stuff there even though it's not optional.

In my opinion it's good this way, because then the svg filename is always `package.svg`, not e.g. `branch.svg`. And it's consistent between services no matter if the stuff before it is optional or not.

Reply:
> In my opinion it's good this way, because then the svg filename is always `package.svg`, not e.g. `branch.svg`. And it's consistent between services no matter if the stuff before it is optional or not.

To be honest I agree with you. It does seem nicer that way.

However these Gem downloads routes, for example, have been established for years:

<img width="775" alt="Screen Shot 2019-05-01 at 12 45 38 PM" src="https://user-images.githubusercontent.com/1487036/57029205-0d012980-6c0f-11e9-84e8-6f01b9365643.png">

All the Code Coverage badges put branch after the repo: https://shields.io/category/coverage

We've done a lot of work with redirects lately to update our routes, though I don't I see being able to do that with these since we'd need a way to detect whether it's branch-first or branch-second.

Since Debian just came online and Ubuntu hasn't yet, and they aren't being used by too many people yet, it might be a good time to change them.

Also, a smaller thing, we're planning to drop the `.svg` or at least make it optional (#2674) which should clean up these routes, too!

Reply:
Fair enough. But how about this: I'll leave it as is here, then let's merge. And if someone wants to change the order for Debian and Ubuntu, they can do it for this too at the same time -- I kind of would not want to spend time that because I still think it would be better the other way around :)

Reply:
Nevermind, moved, see also #3409

Reply:
Thank you!